### DISCUSSION TITLE
How are folks injecting into an episerver page that resides in a class libarary?

### DISCUSSION THREAD

**Discussion by KCRotaryUSA** [episerver/netcore-preview#137] [Q&A]
I have two projects in my solution. One handles the frontend controllers/views and the other library of page/blocks/etc classes (no StartUp class)

If project one has the StartUp that handles connecting to  the EPiServer db and a data db, how are you injecting the data context and things like app data?

I created an extension method in the class libaray and can inject in  a page class, but everytime the page is called from a view, it tries to use a default constructor and not the one with injected values. If I remove the default constructor, I get a RTE because it can't find the default constructor.

**Comment by valdisiljuconoks** [comment]
Do you have any code samples that might speed up answer?

**Reply by KCRotaryUSA** [reply]
I'm working on a sample but this should be pretty straight forward. Create a project with a StartUp and Program/Main. Then create another project that only contains classes (pages/blocks/etc). Then try to inject the dbcontext into those classes and access the dbcontext from a method/property of a class.
